packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts

```diff
@@ -457,25 +457,35 @@ export function updateResolversForIndex(
   }
 
   if (queryField) {
-    makeQueryResolver(config, ctx);
+    makeQueryResolver(config, ctx, dbType);
   }
 
   if (isDynamoDB && syncResolver) {
     makeSyncResolver(name, config, ctx, syncResolver, resolverMap);
   }
 }
 
-function makeQueryResolver(config: IndexDirectiveConfiguration, ctx: TransformerContextProvider) {
+function makeQueryResolver(config: IndexDirectiveConfiguration, ctx: TransformerContextProvider, dbType: DBType) {
+  const { RDSLambdaDataSourceLogicalID } = ResourceConstants.RESOURCES;
+  const isDynamoDB = dbType === DDB_DB_TYPE;
   const { name, object, queryField } = config;
   if (!(name && queryField)) {
     throw new Error('Expected name and queryField to be defined while generating resolver.');
   }
   const modelName = object.name.value;
   const dbInfo = getDBInfo(ctx, modelName);
-  const dataSourceName = `${object.name.value}Table`;
+  let dataSourceName = `${object.name.value}Table`;
+  if (dbType === MYSQL_DB_TYPE) {
+    dataSourceName = RDSLambdaDataSourceLogicalID;
+  }
```

Comment:
As a general pattern I don't love setting a default and then mutating it on conditions. I'd prefer this get extracted to a helper so we can avoid the `let`.

Reply:
+1 here